conda/cli/main_remove.py

```diff
@@ -171,6 +172,13 @@ def execute(args: Namespace, parser: ArgumentParser) -> int:
         msg = "Cannot remove current environment. Deactivate and run conda remove again"
         raise CondaEnvironmentError(msg)
 
+    if args.all and Path(context.target_prefix) == Path(
+        context.default_activation_prefix
+    ):
```

Comment:
Let's reuse the `prefix_data` that was already created above on line 164:

```suggestion
    if args.all and prefix_data == PrefixData(
        context.default_activation_prefix
    ):
```

Reply:
PrefixData's equality is more complete 🤷🏼‍♂️ but also just to mirror the validation pattern from the prior if-clause